app/ts/LiquidityProvidingUI/Liquidity.tsx

```diff
@@ -215,47 +215,57 @@ export const Liquidity = ({ maybeReadClient, maybeWriteClient, universe, reputat
 	const clear = () => {
 		marketData.deepValue = undefined
 	}
+	useSignalEffect(() => {
+		selectedMarket.deepValue
+		clear()
+	})
 
 	useSignalEffect(() => {
-		if (marketAddress.deepValue === undefined) {
-			clear()
-		} else {
-			refreshData()
-		}
+		maybeWriteClient.deepValue
+		maybeReadClient.deepValue
+		selectedMarket.deepValue
```

Comment:
I feel like these 3 lines should be inside of `refreshData`, and they should have a comment indicating why they exist for future readers.